### PR TITLE
lib: anounce modals being opened and closed

### PR DESCRIPTION
**Pull request by KKoukiou**
We need this information in Anaconda, to properly show backdrops.

Relevant to: rhbz#2264015

### CODE REVIEW

**Comment by mvollmer** [comment]
`cockpit-components-dialog` is not only used by storage, but also by some other pages, notably Accounts. Should we restrict this functionality to cockpit-storage?

Also, what about turning this into a flag in sessionStorage? Then there is no risk that Anaconda misses a message.

File: pkg/lib/cockpit-components-dialog.jsx [hunk 1]
```diff
@@ -213,11 +213,20 @@ DialogFooter.propTypes = {
  */
 class Dialog extends React.Component {
     componentDidMount() {
+        // For the scenario that cockpit-storage is used inside anaconda Web UI
+        // We need to know if there is an open dialog in order to create the backdrop effect
+        // on the parent window
+        window.top.postMessage({ cockpitHasModal: true });
```
Comment (mvollmer) [inline pkg/lib/cockpit-components-dialog.jsx:219]: Why not `window.parent`?

Comment (KKoukiou) [reply pkg/lib/cockpit-components-dialog.jsx]: Works as well, fixed.

**Comment by KKoukiou** [comment]
> `cockpit-components-dialog` is not only used by storage, but also by some other pages, notably Accounts. Should we restrict this functionality to cockpit-storage?
> 
> Also, what about turning this into a flag in sessionStorage? Then there is no risk that Anaconda misses a message.

Hmm, I I guess I can also listen on changes in sessionStorage. I have no preference over the other solution, I will adjust.

**Review by KKoukiou — COMMENTED** [review]
Pixel tests are now covering this on the anaconda-webui repo. Can I get a re-review?

**Review by martinpitt — APPROVED** [review]
By far not everything in Cockpit uses cockpit-components-dialog, but the storage page indeed does, and that's what you are interested in. So please go ahead.

**Review by mvollmer — APPROVED** [review]

**Pull request merged**